solr/core/src/java/org/apache/solr/search/SolrDocumentFetcher.java

```diff
@@ -219,6 +221,28 @@ public boolean isLazyFieldLoadingEnabled() {
     }
   }
 
+  /** Returns a collection of the names of all indexed fields which the index reader knows about. */
+  public Collection<String> getIndexedFieldNames() {
+    synchronized (this) {
+      if (indexedFieldNames == null) {
+        indexedFieldNames = new LinkedList<>();
```

Comment:
Let's use an ArrayList instead

Reply:
Yeah I figured you took the same algorithm, hence the LinkedList.  Yes please change to ArrayList there too If you don't mind.

Reply:
Done in https://github.com/apache/solr/pull/757/commits/f4e949e683d60652daeb0e9333cb5fce5a196553 commit.